### PR TITLE
fix: keepkey firmware version check for LTC

### LINKED ISSUE #2374: Filter litecoin accounts based on keepkey firmware version
### Overview

The most recent keepkey firmware version now correctly support segwit and segwit native accounts for litecoin, but previous firmware versions only support legacy accounts.

### References and additional details

https://github.com/keepkey/keepkey-firmware/releases/tag/v7.4.0

### Acceptance Criteria

- Filter out segwit and segwit native (m/49', m/84') on keepkey firmware versions < 7.4.0

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Estimated effort

_No response_

### PR DESCRIPTION
**Pull request by 0xdef1cafe**
## Description

KK 7.4.0 is required for LTC support

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #2374

## Risk

we erroneously skip LTC for a working firmware version

## Testing

**dear ops - plz coordinate amongst yourselves and do NOT update all your KK firmware at once**

plz test with LTC flag **on**

* half of ops update to latest firmware and check LTC works
* half of ops stay on current firmware and ensure that KK works with all other chains with LTC flag
  on, LTC should not show in accounts

## Screenshots (if applicable)

wip

### CODE REVIEW

**Comment by 0xdef1cafe** [comment]
@gomesalexandre requires fleek env vars for keepkey versions url to be updated from `static.shapeshift.com` to the githack url in sample.env

**Review by gomesalexandre — COMMENTED** [review]
Conceptually looks good, will give it a try with two KKs before stamping it

File: src/context/WalletProvider/KeepKey/hooks/useKeepKeyVersions.ts [hunk 1]
```diff
@@ -73,6 +78,10 @@ export const useKeepKeyVersions = () => {
       const latestBootloader = releases.latest.bootloader.version
       const deviceFirmware = await wallet.getFirmwareVersion()
       const latestFirmware = releases.latest.firmware.version
+      if (isLitecoinEnabled) {
+        if (semverMinVersion(MINIMUM_KK_FIRMWARE_VERSION_SUPPORTING_LITECOIN))
```
Comment (0xdef1cafe) [inline src/context/WalletProvider/KeepKey/hooks/useKeepKeyVersions.ts:82]: @gomesalexandre this check isn't working

File: src/context/AppProvider/AppContext.tsx [hunk 2]
```diff
@@ -149,6 +156,16 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
...
               if (!supportsBTC(wallet)) continue
 
+              if (
+                chainReference === CHAIN_REFERENCE.LitecoinMainnet &&
+                wallet instanceof KeepKeyHDWallet &&
+                isLitecoinEnabled &&
+                versions // without versions, we can't be sure the isLTCSupportedFirmwareVersion is valid
+              ) {
+                // skip litecoin if KK firmware update is required
+                if (!isLTCSupportedFirmwareVersion) continue
```
Comment (kaladinlight) [inline src/context/AppProvider/AppContext.tsx:166]: Does product want all or nothing? We would have the option of showing legacy litecoin accounts on old firmware version if we wanted

File: src/context/WalletProvider/KeepKeyProvider.tsx [hunk 3]
```diff
@@ -178,6 +180,9 @@ export const KeepKeyProvider = ({ children }: { children: React.ReactNode }): JS
                   <AlertTitle>{translate('updateToast.keepKey.title')}</AlertTitle>
                   <AlertDescription>
                     <Text>{translate('updateToast.keepKey.newUpdateAvailable')}</Text>
+                    {isLitecoinEnabled && !isLTCSupportedFirmwareVersion ? (
+                      <Text>This update is required to support Litecoin</Text>
```
[the remark this answers is not in the document]
Comment (gomesalexandre) [reply src/context/WalletProvider/KeepKeyProvider.tsx]: https://github.com/shapeshift/web/pull/2385/commits/0611694eef63b34a3aefade26a76b8dec20e0063

**Comment by cjthompson** [comment]
Doesn't this happen on the hdwallet side? hdwallet includes code for checking if a chain is supported and can check the firmware version.

File: src/context/AppProvider/AppContext.tsx (hunk 2 above)
Comment (0xdef1cafe) [reply src/context/AppProvider/AppContext.tsx:166]: @kaladinlight confirmed per @DiggyDiggy2 we want the "all or nothing" approach of forcing them to update firmware given we waited. partial support can and will cause confusion.

**Review by gomesalexandre — APPROVED** [review]
🍬

**Review by elmutt — DISMISSED** [review]

**Review by gomesalexandre — APPROVED** [review]

**Pull request merged**